go/pkg/db/traced_tx.go

```
@@ -0,0 +1,174 @@
+
+// QueryContext executes a SQL query within the transaction with tracing
+func (t *TracedTx) QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error) {
+	ctx, span := tracing.Start(ctx, "Tx.QueryContext")
+	defer span.End()
+	span.SetAttributes(
+		attribute.String("mode", t.mode),
+		attribute.String("query", query),
+	)
+
+	start := time.Now()
+	rows, err := t.tx.QueryContext(ctx, query, args...)
```

golangci-lint (2.2.2)

Rows/Stmt/NamedStmt was not closed

(sqlclosecheck)